miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -817,12 +834,19 @@ protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) throws
       config.put(Config.SORTABLE_LOCATION, false);
       config.put(Config.PROPAGATE, false);
       config.put(Config.IS_LIBRARY_RECEIPT, true);
+      config.putPOJO(Config.BOX, newBox);
     }
 
   }
 
-  private final BulkPropagateTableBackend<Library, DilutionDto> dilutionBulkPropagateBackend = new BulkPropagateTableBackend<Library, DilutionDto>(
-      "dilution", DilutionDto.class, "Dilutions", "Libraries") {
+  private final class BulkPropagateLibraryBackend extends BulkPropagateTableBackend<Library, DilutionDto> {
+
+    private final BoxDto newBox;
+
+    private BulkPropagateLibraryBackend(BoxDto newBox) {
+      super("dilution", DilutionDto.class, "Dilutions", "Libraries");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Dilutions" 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)